tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -429,6 +429,8 @@ internal SpanContext CreateSpanContext(ISpanContext parent = null, string servic
                     traceContext.Origin = parentSpanContext.Origin;
                     traceContext.AdditionalW3CTraceState = parentSpanContext.AdditionalW3CTraceState;
                 }
+
+                lastParentId = parentSpanContext.LastParentId;
```

Comment:
I think this will cause _all_ spans in a trace to inherit this value from their parent, so every span in the trace will have a value in `span.Context.LastParentId` (if we extracted context from `traceparent`), so all spans in that trace will be encoded with the `_dd.parent_id` tag in `meta`.

But if I understand the RFC, we only want the local root span to have this tag.

Reply:
Yes good catch thanks.
I wrapped it in a `if (parentSpanContext.IsRemote)` check which I think should address that so only spans coming off of that extracted context will get the `_dd.parent_id` tag.